js/components/address.js

```diff
@@ -0,0 +1,298 @@
+
+      fields.forEach(field => {
+        if (selectedValues.includes(field.name)) {
+          field[attr] = value;
+        }
+      });
+    },
+    updateDisabledOptions: function() {
+      var selectedValues = Object.values(this.selectedFieldOptions);
+
+      this.fieldOptions.forEach(option => {
+        if (selectedValues.includes(option.label)) {
+          option.disabled = true;
+        } else {
+          option.disabled = false;
+        }
+      });
+    },
+    resetOptionsOnSelectOne: function(key) {
+      if (this.selectedFieldOptions[key] === '') {
+        this.updateDisabledOptions();
+      }
+    },
+    initAutocomplete: async function(input, countryRestrictions) {
+      this.addressField = Fliplet.UI.AddressField(this.$refs.addressField);
+      await this.addressField.getAutocompleteSuggestions(input, countryRestrictions)
+        .then(suggestions => {
+          if (typeof this.value === 'object') {
+            this.addressSuggestions = [];
+            this.suggestionSelected = true;
+          } else {
+            this.addressSuggestions = suggestions;
+          }
+        });
+    },
+    onChange: function() {
+      var $vm = this;
+
+      this.addressField.change((value) => {
+        if (this.addressComponents.length) {
+          this.assignValuesToSeparateFields(this.addressComponents, this.separateFieldsName);
+
+          for (const key in this.selectedFieldOptions) {
+            if (this.selectedFieldOptions[key]) {
+              var matchedField = this.separateFieldsName.find(field => field.key === key);
+              var fieldName = this.selectedFieldOptions[key];
+
+              if (matchedField) {
+                (function(fieldName, matchedFieldValue) {
+                  Fliplet.FormBuilder.get()
+                    .then(function(form) {
+                      form.field(fieldName).set(matchedFieldValue);
+                    });
+                })(fieldName, matchedField.value);
+              }
+            }
+          }
+        }
+
+        this.updateSelectedFieldsProperty('readonly', this.manualInput);
+
+
+        if (!$vm.manualInput && this.addressComponents.length) {
+          this.suggestionSelected = true;
+          this.value = value;
+          this.updateValue();
+        } else {
+          this.value = value;
+          this.updateValue();
+        }
+
+        this.suggestionSelected = true;
+      });
+    },
+    onBeforeSubmit: function(data) {
+      var $vm = this;
+      var value = $vm.lastChosenAutocompleteValue;
+
+      if (!this.manualInput) {
+        if (data.hasOwnProperty(this.name)) {
+          data[this.name] = value;
+        }
+      }
+
+      this.updateSelectedFieldsProperty('value', '');
+    }
+  },
+  watch: {
+    value: function(val) {
+      var $vm = this;
+
+      if (typeof val === 'object') {
+        $vm.lastChosenAutocompleteValue = val.label;
+        $vm.suggestionSelected = true;
+        val = val.label;
```

Comment:
@armine-fliplet, there's no need for shadowing `this` with `$vm.` You don't pass it any further. Please use `this` instead.